File: src/main/java/de/tectoast/emolga/utils/sql/base/columns/BooleanColumn.java
package de.tectoast.emolga.utils.sql.base.columns;

import de.tectoast.emolga.utils.sql.base.DataManager;
import org.jetbrains.annotations.Nullable;

import java.sql.ResultSet;
import java.sql.SQLException;

public class BooleanColumn extends SQLColumn<Boolean> {

    public BooleanColumn(String name, DataManager manager) {
        super(name, manager);
    }

    @Override
    public @Nullable Boolean getValue(ResultSet set) {
        try {
            return set.getBoolean(name);
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return null;
    }

    @Override
    public Boolean retrieveValue(SQLColumn<?> checkcolumn, Object checkvalue) {
        return DataManager.read(manager.select(checkcolumn.check(checkvalue), this), rs -> {
            if (rs.next()) {
                return getValue(rs);
            }
            return false;
        });
    }

    @Override
    public String wrap(Object value) {
        return value.equals(true) ? "TRUE" : "FALSE";
    }
}

File: src/main/kotlin/de/tectoast/emolga/utils/sql/base/columns/BooleanColumn.kt
package de.tectoast.emolga.utils.sql.base.columns

import de.tectoast.emolga.utils.sql.base.DataManager
import java.sql.ResultSet

class BooleanColumn(name: String, manager: DataManager) : SQLColumn<Boolean>(name, manager) {
    override fun getValue(set: ResultSet): Boolean {
        return set.getBoolean(name)
    }

    override fun retrieveValue(checkcolumn: SQLColumn<*>, checkvalue: Any): Boolean? {
        return DataManager.read<Boolean?>(manager.select(checkcolumn.check(checkvalue), this)) { rs: ResultSet ->
            if (rs.next()) getValue(rs) else
                null
        }
    }

    override fun wrap(value: Any?): String {
        return if (value == true) "TRUE" else "FALSE"
    }
}